apps/voting/app/src/screens/Votes.js

```diff
@@ -1,17 +1,296 @@
-import React from 'react'
+import React, { useCallback, useState, useEffect } from 'react'
+import {
+  BackButton,
+  Bar,
+  DropDown,
+  GU,
+  Tag,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { isWithinInterval, startOfDay, endOfDay } from 'date-fns'
 import VotingCard from '../components/VotingCard/VotingCard'
 import VotingCardGroup from '../components/VotingCard/VotingCardGroup'
+import Vote from '../components/Vote'
+import EmptyFilteredVotes from '../components/EmptyFilteredVotes'
+import DateRangeInput from '../components/DateRange/DateRangeInput'
+import {
+  VOTE_STATUS_REJECTED,
+  VOTE_STATUS_ACCEPTED,
+  VOTE_STATUS_ONGOING,
+  VOTE_STATUS_EXECUTED,
+} from '../vote-types'
+import { getVoteStatus, isVoteAction } from '../vote-utils'
+import { useSettings } from '../vote-settings-manager'
 
-const Votes = React.memo(({ votes, onSelectVote }) => {
-  const sortedVotes = votes.sort((a, b) => {
-    const dateDiff = b.data.endDate - a.data.endDate
-    // Order by descending voteId if there's no end date difference
-    return dateDiff !== 0 ? dateDiff : b.voteId - a.voteId
+const sortVotes = (a, b) => {
+  const dateDiff = b.data.endDate - a.data.endDate
+  // Order by descending voteId if there's no end date difference
+  return dateDiff !== 0 ? dateDiff : b.voteId - a.voteId
+}
+
+const useFilterVotes = votes => {
```

Comment:
Can you help me understand the benefit of merging `useFilteredVotes` with `useVotes`? I am inclined on having them independent as I see filtering an effect happening outside of votes, easier to decouple, not tied together (hope the image I'm picturing was correctly communicated).

Reply:
Quick update here: I had to move `useFilterVotes` one component up (`App`) to make the filters persist when the user navigates to a vote and back.

Reply:
> The benefit of merging useFilteredVotes with useVotes

I had the image of just doing the filtering inside `useVotes()`, so that it was just one hook to produce the rendered array of votes. In this particular case I see it as:

```
input votes --> transform for rendering (hook: filter, split) --> render
```

And it's easier if that transform was done in a single hook to ease debugging and reasoning about the code later.

But no particular strong feelings either way, I'd just lean towards less hook definitions than more, especially if they end up manipulating the same state.